test/over_react_redux/connect_test.dart

```diff
@@ -0,0 +1,444 @@
+      });
+    });
+
+    group('mergeProps', (){
+      test('properly merges props based on consumer map', () async {
+        ConnectedCounter = connect<CounterState, CounterProps>(
+          mapStateToProps: (state){
+            return Counter()..currentCount = state.count;
+          },
+          mapDispatchToProps: (dispatch){
+            return Counter()..dispatchDecrement = () => dispatch(DecrementAction());
+          },
+          mergeProps: (stateProps, dispatchProps, ownProps) {
+            expect(stateProps, isA<CounterProps>());
+            expect(dispatchProps, isA<CounterProps>());
+            expect(ownProps, isA<CounterProps>());
+            return Counter()
+                // Return whatever value is passed through ownProps until the state count is over 1
+                ..currentCount = stateProps.currentCount < 1 ? ownProps.currentCount : stateProps.currentCount
+                ..dispatchDecrement = ownProps.dispatchDecrement;
+          },
+          forwardRef: true,
+        )(Counter);
+
+        jacket = mount(
+          (ReduxProvider()..store = store1)(
+            (ConnectedCounter()
+                ..ref = (ref){ counterRef = ref; }
+                // make `dispatchDecrement` increment
+                ..dispatchDecrement = () {store1.dispatch(IncrementAction());}
+                ..currentCount = 900
+            )('test'),
+          ),
+        );
+        // `button-decrement` will be incrementing now
+        var dispatchButton = getByTestId(jacket.getInstance(), 'button-decrement');
+
+        expect(getDartComponent<CounterComponent>(counterRef).props.dispatchDecrement, isA<Function>());
+
+        // state.count is at 0
+        expect(getDartComponent<CounterComponent>(counterRef).props.currentCount, 900);
+        expect(jacket.getNode().innerHtml, contains('Count: 900'));
+
+        // Click button mapped to trigger `propFromDispatch` prop.
+        click(dispatchButton); // state.count is now equal to 1
+
+        // wait for the next tick for the async dispatch to propagate
+        await Future(() {});
+
+        expect(getDartComponent<CounterComponent>(counterRef).props.currentCount, 1);
+        expect(jacket.getNode().innerHtml, contains('Count: 1'));
+      });
+    });
+
+    group('Options', () {
+      group('areOwnPropsEqual', (){
+        test('', () {
+          ConnectedCounter = connect<CounterState, CounterProps>(
+            areOwnPropsEqual: expectAsync2((next, prev) {
+              expect(next, isA<CounterProps>());
+              expect(prev, isA<CounterProps>());
+              expect(next.id, 'test');
+              expect(prev.id, 'test2');
+              return true;
+            }),
+          )(Counter);
+
+          var whatever = connectOptions.areOwnPropsEqual(
+            JsBackedMap.from({'id':'test'}).jsObject,
+            JsBackedMap.from({'id':'test2'}).jsObject
+          );
+
+          expect(whatever, true);
```

Comment:
#nit
```suggestion
          expect(whatever, isTrue);
```
Also, `expect whatever is true` lol